documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,354 @@
+Radix Sort
+##########
+
+-----------------------------------------------------------
+``radix_sort`` and ``radix_sort_by_key`` Function Templates
+-----------------------------------------------------------
+
+The ``radix_sort`` and ``radix_sort_by_key`` functions sort data using the radix sort algorithm.
+The sorting is stable, ensuring the preservation of the relative order of elements with equal keys.
+The functions implement Onesweep* [#fnote1]_ algorithm variant.
+
+A synopsis of the ``radix_sort`` and ``radix_sort_by_key`` functions is provided below:
+
+.. code:: cpp
+
+   // defined in <oneapi/dpl/experimental/kernel_templates>
+
+   namespace oneapi::dpl::experimental::kt::esimd {
+
+   template <bool IsAscending = true, std::uint8_t RadixBits = 8, typename KernelParam, typename Range>
+   sycl::event
+   radix_sort(sycl::queue q, Range&& rng, KernelParam param); // (1)
+
+   template <bool IsAscending = true,  std::uint8_t RadixBits = 8, typename KernelParam, typename Iter>
+   sycl::event
+   radix_sort(sycl::queue q, Iter first, Iter last, KernelParam param); // (2)
+
+   template <bool IsAscending = true, std::uint8_t RadixBits = 8, typename KernelParam, typename KeysRng, typename ValsRng>
+   sycl::event
+   radix_sort_by_key(sycl::queue q, KeysRng&& keys_rng, ValsRng&& vals_rng, KernelParam param); // (3)
```

Comment:
```suggestion
   template <bool IsAscending = true, std::uint8_t RadixBits = 8,
             typename KernelParam, typename Range1, typename Range2>
   sycl::event
   radix_sort_by_key (sycl::queue q, Range1&& keys, Range2&& values, KernelParam param); // (3)
```
Similarly for (4)

Reply:
I've also split the function to avoid scrolling. The proposed line slightly bigger than necessary.